src/back/hlsl/writer.rs

```diff
@@ -1388,3 +1410,97 @@ fn scalar_kind_str(kind: crate::ScalarKind, width: crate::Bytes) -> Result<&'sta
         Sk::Bool => Ok("bool"),
     }
 }
+
+fn storage_access(storage_access: crate::StorageAccess) -> Option<&'static str> {
+    if storage_access == crate::StorageAccess::LOAD {
+        Some("ByteAddressBuffer")
+    } else if storage_access.is_all() {
+        Some("RWByteAddressBuffer")
+    } else {
+        None
+    }
+}
+
+/// Helper function that return string with default zero initialization for supported types
+fn zero_init_value_str(inner: &TypeInner) -> Option<String> {
```

Comment:
Can we just use the C-style initialization:
```cpp
Something foo = {}; // initializes to zeroes automatically
```

Reply:
You mean for `struct`? Currently this code initialize only scalar, vector and matrix.

Reply:
I assume `= {}` works for all types in HLSL just like it does in C.

Reply:
TIL, Thank you! Fixed.

Reply:
Why do we still need this function then?

Reply:
This not working for hlsl:
```
./shader.hlsl:7:14: error: too few elements in vector initialization (expected 2 elements, have 0)
  float2 v = {};
```

Based on your suggestion I reworked from:
```hlsl
float2 v = float2(0.0, 0.0);
```
To:
```hlsl
float2 v = { 0.0, 0.0 };
```

Reply:
Apologies, I confused you with this advice! HLSL has zero initialization semantics but with a different syntax from C:
```
Foo foo = (Foo)0;
```
This should work on all types.

Reply:
Thank you! I think it's finally fixed :smile:

Reworked this function and now it's write directly to buffer to avoid heap allocation.